nix/devshell.nix

```diff
@@ -1,59 +1,73 @@
-{inputs, ...}: {
-  imports = [
-    inputs.devshell.flakeModule
-  ];
+{
+  pkgs,
+  perSystem,
+  ...
+}: let
+  inherit (pkgs) lib;
+  inherit (pkgs.stdenv) isLinux isDarwin;
+in
+  perSystem.devshell.mkShell {
+    env = [
+      {
+        name = "GOROOT";
+        value = pkgs.go + "/share/go";
+      }
+      {
+        name = "LD_LIBRARY_PATH";
+        value = "$DEVSHELL_DIR/lib";
+      }
+    ];
 
-  config.perSystem = {
-    lib,
-    pkgs,
-    config,
-    ...
-  }: let
-    inherit (pkgs.stdenv) isLinux isDarwin;
-  in {
-    config.devshells.default = {
-      env = [
-        {
-          name = "GOROOT";
-          value = pkgs.go + "/share/go";
-        }
-        {
-          name = "LD_LIBRARY_PATH";
-          value = "$DEVSHELL_DIR/lib";
-        }
-      ];
+    packages = lib.mkMerge [
+      (with pkgs; [
+        # golang
+        go
+        goreleaser
+        golangci-lint
+        delve
+        pprof
+        graphviz
 
-      packages = lib.mkMerge [
-        (with pkgs; [
-          # golang
-          go
-          goreleaser
-          golangci-lint
-          delve
-          pprof
-          graphviz
-        ])
-        # platform dependent CGO dependencies
-        (lib.mkIf isLinux [
-          pkgs.gcc
-        ])
-        (lib.mkIf isDarwin [
-          pkgs.darwin.cctools
-        ])
-        # include formatters for development and testing
-        (import ./formatters.nix pkgs)
-      ];
+        # docs
+        nodejs
+      ])
+      # platform dependent CGO dependencies
+      (lib.mkIf isLinux [pkgs.gcc])
+      (lib.mkIf isDarwin [pkgs.darwin.cctools])
```

Comment:
I think this can be replaced by  `pkgs.stdenv.cc.cc`. It resolves to clang on macOS and GCC on Linux.

Reply:
Good to know. I went ahead and removed CGO dependencies and disabled it as we don't need it anyway, this was a copy past from another project.